linux/src/async_socket_linux.c

```diff
@@ -514,30 +517,24 @@ int async_socket_open_async(ASYNC_SOCKET_HANDLE async_socket, ON_ASYNC_SOCKET_OP
         // Codes_SRS_ASYNC_SOCKET_LINUX_11_029: [ If async_socket is already OPEN or OPENING, async_socket_open_async shall fail and return a non-zero value. ]
         if (current_state != ASYNC_SOCKET_LINUX_STATE_CLOSED)
         {
+            // Codes_SRS_ASYNC_SOCKET_LINUX_11_034: [ If any error occurs, async_socket_open_async shall fail and return a non-zero value. ]
             LogError("Open called in state %" PRI_MU_ENUM "", MU_ENUM_VALUE(ASYNC_SOCKET_LINUX_STATE, current_state));
             result = MU_FAILURE;
         }
         else
         {
-            // Codes_SRS_ASYNC_SOCKET_LINUX_11_030: [ If async_socket has already closed the underlying socket handle then async_socket_open_async shall fail and return a non-zero value. ]
-            if (async_socket->socket_handle == INVALID_SOCKET)
-            {
-                // Codes_SRS_ASYNC_SOCKET_LINUX_11_034: [ If any error occurs, async_socket_open_async shall fail and return a non-zero value. ]
-                LogError("Open called after socket was closed");
-                result = MU_FAILURE;
-            }
-            else
-            {
-                // Codes_SRS_ASYNC_SOCKET_LINUX_11_031: [ async_socket_open_async shall add the socket to the epoll system by calling epoll_ctl with EPOLL_CTL_ADD. ]
-                // Codes_SRS_ASYNC_SOCKET_LINUX_11_032: [ async_socket_open_async shall set the state to OPEN. ]
-                (void)interlocked_exchange(&async_socket->state, ASYNC_SOCKET_LINUX_STATE_OPEN);
-                // Codes_SRS_ASYNC_SOCKET_LINUX_11_033: [ On success async_socket_open_async shall call on_open_complete_context with ASYNC_SOCKET_OPEN_OK. ]
-                on_open_complete(on_open_complete_context, ASYNC_SOCKET_OPEN_OK);
-
-                // Codes_SRS_ASYNC_SOCKET_LINUX_11_028: [ On success, async_socket_open_async shall return 0. ]
-                result = 0;
-                goto all_ok;
-            }
+            async_socket->socket_handle = socket_handle;
```

Comment:
>socket_handle [](http://example.com/codeflow?start=42&length=13)

Would be nice to have a structure like OPEN_DATA, where the data relevant for the OPEN state is kept. That way it is harder to inadvertently access it from create/destroy (not impossible, but would require the developer to write a little bit more). Also later it is wasier to make the OPEN_DATA a handle by itself.

Given the time constraints and the fact that this would be squirrling a bit, I give it a consider, but I did notice it, so "see something say something"

Reply:
I will make a task for this, I would like to do more here and I have ideas, but due to time constraints a task will do for now.

Reply:
Task here: https://msazure.visualstudio.com/One/_workitems/edit/28084746